src/lib/src/component/tree-viewer/tree-viewer.component.events.spec.ts

```diff
@@ -0,0 +1,131 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { DebugElement } from '@angular/core';
+import { By } from '@angular/platform-browser';
+
+import { MessagingService } from '@testeditor/messaging-service';
+
+import { TreeViewerComponent } from './tree-viewer.component';
+import { WorkspaceElement } from '../../service/persistence/workspace-element';
+import * as events from './event-types';
+import { testBedSetup } from './tree-viewer.component.spec';
+
+describe('TreeViewerComponent event handling', () => {
+
+  let component: TreeViewerComponent;
+  let fixture: ComponentFixture<TreeViewerComponent>;
+  let messagingService: MessagingService;
+
```

Comment:
That's a valid concern but luckily TypeScript got us covered. We compile to ES5 here so the following code in TypeScript:
```
let isOwnElement = element.path === this.model.path;
```
compiles to:
```
var isOwnElement = element.path === _this.model.path;
```